src/main/groovy/com/brightsparklabs/gradle/baseline/BaselinePlugin.groovy

```diff
@@ -351,45 +358,139 @@ public class BaselinePlugin implements Plugin<Project> {
 
         // Error early if configuration in invalid.
         if (bucketNameIsEmpty) {
-            def error = "`bslGradle.deploy.s3.bucketName` cannot be null or empty. Value was: `${bucketName}`"
+            def error = "`bslGradle.deploy.s3.bucketName` cannot be null or empty. Value was:" +
+                    " `${bucketName}`"
             project.logger.error(error)
             throw new IllegalStateException(error)
         }
         if (filesToUploadIsEmpty) {
-            def error = "`bslGradle.deploy.s3.filesToUpload` cannot be null or empty. Value was: ${filesToUpload}"
+            def error = "`bslGradle.deploy.s3.filesToUpload` cannot be null or empty. Value was:" +
+                    " ${filesToUpload}"
             project.logger.error(error)
             throw new IllegalStateException(error)
         }
 
+        // Get the absolute paths of the files to upload. We search for files in the `build`
```

Comment:
This whole section where you calculate the files to upload should be in the `doLast{}` block not in the setup block.
The way you have it at the moment means you would get different results if you have or have not called a build before hand.

I suggest you make the task depend on `build` as a minimum. People can add additional targets in the `build.gradle` file if desired.